source/server/ssl_context_manager.cc

```diff
@@ -0,0 +1,40 @@
+#include "server/ssl_context_manager.h"
+
+#include "envoy/common/exception.h"
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Server {
+
+class SslContextManagerStub final : public Envoy::Ssl::ContextManager {
+  Ssl::ClientContextSharedPtr
+  createSslClientContext(Stats::Scope& /* scope */,
+                         const Envoy::Ssl::ClientContextConfig& /* config */) override {
+    throw EnvoyException("SSL is not supported in this configuration");
+  }
+
+  Ssl::ServerContextSharedPtr
+  createSslServerContext(Stats::Scope& /* scope */,
+                         const Envoy::Ssl::ServerContextConfig& /* config */,
+                         const std::vector<std::string>& /* server_names */) override {
+    throw EnvoyException("SSL is not supported in this configuration");
+  }
+
+  size_t daysUntilFirstCertExpires() const override { return std::numeric_limits<int>::max(); }
```

Comment:
Shouldn't every function in here throw for consistency? Can you delegate the throw to a private helper function so we don't have a duplicated error message string?

Reply:
I think the code is bit simpler if there's such a semi-functional stub because `Envoy::Server::ServerStats` and `Envoy::Server::Admin` rely on a non-null SSL context manager to report on the server's certificate data even if no TLS context is configured like in https://github.com/envoyproxy/envoy/blob/ac21376c65c69d565e51f6b6570f20671d9e6416/source/server/server.cc#L196 and https://github.com/envoyproxy/envoy/blob/ac21376c65c69d565e51f6b6570f20671d9e6416/source/server/http/admin.cc#L1227.

Reply:
OK that's fine.